pkg/sampler/sampler.go

```diff
@@ -33,15 +35,13 @@ func (t *Target) SetHash() {
 
 type Sample struct {
 	StatusCode      int
-	T1              time.Time
-	T2              time.Time
-	ResponseHeaders map[string]string
-}
-
-// Latency returns the amount of milliseconds between T1
-// and T2 (start and finish).
-func (s Sample) Latency() float64 {
-	return s.T2.Sub(s.T1).Seconds() * 1000
+	TimeStart       time.Time
+	TimeToConnect   time.Time
+	TimeToFirstByte time.Time
+	TimeEnd         time.Time
```

Comment:
I can't ever imagine wanting to know the actual time of day that the first byte arrived.  I can sort of see maybe wanting to calculate the total time to first byte, including connect?  But if you're going to split these out, I just think it makes more sense to make these deltas from the previous phase.